app/assets/stylesheets/components/_nav-side.scss

```diff
@@ -37,29 +37,23 @@ $side-nav-style-2-label: rgba($grey, .8);
 .nav {
   &--side {
     @include nav-side-style($side-nav-style-1-button, $side-nav-style-1-label);
-
     position: fixed;
     top: 50%;
     right: 20px;
     z-index: $z-200;
 
     transform: translateY(-50%);
 
-    &.nav--forewords {
-      @include nav-side-style($side-nav-style-2-button, $side-nav-style-2-label);
-    }
-
-    &.nav--foreword {
-      @include nav-side-style($side-nav-style-1-button, $side-nav-style-1-label);
-    }
-
-    &.nav--projects,
-    &.nav--highlights {
-      @include nav-side-style($side-nav-style-2-button, $side-nav-style-2-label);
-    }
+    @for $index from 1 through 4 {
+      $point-of-change: nth(('--forewords', '--foreword', '--projects', '--stats-people'), $index);
```

Comment:
Yep, I definitely agree with this. I think the for loop is what pushes it over the edge into being overly complex.

However, in refactoring, I did come across the original reason that I chose to do this; because the nav scrolling classes are accumulative as you scroll down the page, the order that the styling rules are written in is important. So to achieve the same effect as this loop, it's necessary to write four separate rules, in the correct order, as seen in the screenshot...

I do agree that the loop is a bit difficult to read, so I've switched it back to the four separate rules, but maybe there's some kind of middle ground that could be found? Or do you think we should avoid logic in Sass as much as we can?

<img width="1581" alt="image" src="https://user-images.githubusercontent.com/56026099/126297553-060f45de-a5b9-4463-8cbd-944a12009e9b.png">

Reply:
Yeah I think we should avoid it --- if only to keep sections like this as easy to read as possible. I see the use of more complex SASS logic as reserved to utility functions where they really make the most sense depending on parameters. We want to have a low figuring out time. When we reach for things like loops it implies our code is going to be reused, but this is only run once.

It absolutely, on the other hand, makes sense to write in comments where ever possible to  give warnings to other developers if the ordering is particularly important.

Reply:
Cool. I'll certainly put in a comment - maybe a future dev can find a way of DRYing this up which doesn't diminish readability.